packages/runtime/src/runtime.ts

```diff
@@ -646,8 +647,11 @@ export class Runtime {
    * each opcode execution (upto depth = debugStack)
    * NOTE: Application mode is only supported in TEALv > 1
    */
-  run (program: string, executionMode: ExecutionMode, debugStack?: number): void {
+  run (program: string, executionMode: ExecutionMode, indexInGroup: number, debugStack?: number): void {
     const interpreter = new Interpreter();
     interpreter.execute(program, executionMode, this, debugStack);
+    if (executionMode === ExecutionMode.APPLICATION) {
+      this.ctx.sharedScratchSpace.set(indexInGroup, interpreter.scratch);
+    }
```

Comment:
Can we do it inside interpreter only (after executing all opcodes)? Could you please check?
Inside interpreter we have access to ctx (`this.runtime.ctx`). We can also get `indexInGroup` (using `ctx.gtxs`).

If possible then we won't need to pass index in each `run`.

Reply:
Good idea, I will try this.

Reply:
i checked it, to get indexInGroup from ctx.gtxs there are two options 
1. traverse gtxs and match tx(this seems not good  and slow) 
2. have a tx variable set in runtime but this will be similar to what we have
3. I like current implementation because it is easy to understand .

Reply:
yes, i was thinking the first one (`indexInGroup = gtxs.indexOf(tx);`). It introduces an extra loop. 
Ok, let's keep what we have.